pbs_light.go

```diff
@@ -378,26 +378,51 @@ func (deps *auctionDeps) auction(w http.ResponseWriter, r *http.Request, _ httpr
 		}
 	}
 	if pbs_req.CacheMarkup == 1 {
-		cobjs := make([]*pbc.CacheObject, len(pbs_resp.Bids))
-		for i, bid := range pbs_resp.Bids {
-			bc := &pbc.BidCache{
-				Adm:    bid.Adm,
-				NURL:   bid.NURL,
-				Width:  bid.Width,
-				Height: bid.Height,
+		var cobjs []*pbc.CacheObject
+		var cxmls []*pbc.CacheXmlObject
+		for _, bid := range pbs_resp.Bids {
+			if bid.CreativeMediaType == "video" {
+				cxmls = append(cxmls, &pbc.CacheXmlObject{
+					Value: bid.Adm,
+				})
+			} else {
+				bc := &pbc.BidCache{
+					Adm:    bid.Adm,
+					NURL:   bid.NURL,
+					Width:  bid.Width,
+					Height: bid.Height,
+				}
+				cobjs = append(cobjs, &pbc.CacheObject{
+					Value: bc,
+				})
 			}
-			cobjs[i] = &pbc.CacheObject{
-				Value: bc,
+		}
+		if len(cobjs) > 0 {
+			err = pbc.Put(ctx, cobjs)
+			if err != nil {
```

Comment:
I think this and this (https://github.com/prebid/prebid-server/pull/283/files#diff-444feda808b76687bcc8827c1025876cR410) can be condensed into one check since output is the same.

Reply:
What if the first Put method returns error but the second PutXml doesn't return error, how should them get condensed?